### ISSUE TITLE
Windows support?

### ISSUE THREAD

**Issue by DmitrySoshnikov** [mbrt/librsync-rs#14]
Trying to compile the basic example from the docs on Windows, fails in liner. Not sure if it's specifics of my Windows configuration, or whether this crate doesn't have Windows support yet. The original librsync should build normally on Windows.

The crate compiles normally on Mac OS with correct delta files generation. On Linux (Ubuntu server) I noticed different file size generated for a delta patch. And on Windows cannot build yet.

```
librsync/src/checksum.c(28): error C2491: 'RS_MD4_SUM_LENGTH': definition of dllimport data not allowed
librsync/src/checksum.c(29): error C2491: 'RS_BLAKE2_SUM_LENGTH': definition of dllimport data not allowed
```

**Comment by mbrt** [comment]
Yeah, I fear this was broken with the last updates. The problem is that I no longer have access to a Windows machine, so I can't easily check Windows support.

Could you follow the steps in https://github.com/mbrt/librsync-rs/blob/master/librsync-sys/README.md and see whether you get different generated headers for Windows?

**Comment by changhe3** [comment]
I got this:
```
PS C:\Users\admin\Documents\GitHub\librsync\gen> cargo tree
librsync-gen v0.1.0 (C:\Users\admin\Documents\GitHub\librsync\gen)
[build-dependencies]
└── cmake v0.1.12
    └── gcc v0.3.21
        ├── advapi32-sys v0.1.2
        │   └── winapi v0.2.5
        │   [build-dependencies]
        │   └── winapi-build v0.1.1
        └── winapi v0.2.5
PS C:\Users\admin\Documents\GitHub\librsync\gen> cargo run
    Updating crates.io index
   Compiling cc v1.0.69
   Compiling cmake v0.1.45
   Compiling librsync-gen v0.1.0 (C:\Users\admin\Documents\GitHub\librsync\gen)
error: failed to run custom build command for `librsync-gen v0.1.0 (C:\Users\admin\Documents\GitHub\librsync\gen)`

Caused by:
  process didn't exit successfully: `C:\Users\admin\Documents\GitHub\librsync\gen\target\debug\build\librsync-gen-876be86ab9e95625\build-script-build` (exit code: 101)
  --- stdout
  running: "cmake" "C:\\Users\\admin\\Documents\\GitHub\\librsync\\gen\\.." "-G" "Visual Studio 16 2019" "-Thost=x64" "-Ax64" "-DBUILD_RDIFF=OFF" "-DENABLE_COMPRESSION=OFF" "-DCMAKE_INSTALL_PREFIX=C:\\Users\\admin\\Documents\\GitHub\\librsync\\gen\\target\\debug\\build\\librsync-gen-02a144ff907295ec\\out" "-DCMAKE_C_FLAGS= -nologo -MD -Brepro" "-DCMAKE_C_FLAGS_DEBUG= -nologo -MD -Brepro" "-DCMAKE_CXX_FLAGS= -nologo -MD -Brepro" "-DCMAKE_CXX_FLAGS_DEBUG= -nologo -MD -Brepro" "-DCMAKE_ASM_FLAGS= -nologo -MD -Brepro" "-DCMAKE_ASM_FLAGS_DEBUG= -nologo -MD -Brepro" "-DCMAKE_BUILD_TYPE=Debug"
  -- Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.17763.
  -- The C compiler identification is MSVC 19.29.30040.0
  -- Detecting C compiler ABI info
  -- Detecting C compiler ABI info - done
  -- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/BuildTools/VC/Tools/MSVC/14.29.30037/bin/Hostx64/x64/cl.exe - skipped
  -- Detecting C compile features
  -- Detecting C compile features - done
  -- DO_RS_TRACE=0
  -- Looking for include file sys/file.h
  -- Looking for include file sys/file.h - not found
  -- Looking for include file sys/stat.h
  -- Looking for include file sys/stat.h - found
  -- Looking for include file sys/types.h
  -- Looking for include file sys/types.h - found
  -- Looking for include file unistd.h
  -- Looking for include file unistd.h - not found
  -- Looking for include file fcntl.h
  -- Looking for include file fcntl.h - found
  -- Looking for include file mcheck.h
  -- Looking for include file mcheck.h - not found
  -- Looking for include file zlib.h
  -- Looking for include file zlib.h - not found
  -- Looking for include file bzlib.h
  -- Looking for include file bzlib.h - not found
  -- Looking for __func__
  -- Looking for __func__ - found
  -- Looking for __FUNCTION__
  -- Looking for __FUNCTION__ - found
  -- Looking for fseeko
  -- Looking for fseeko - not found
  -- Looking for fseeko64
  -- Looking for fseeko64 - not found
  -- Looking for _fseeki64
  -- Looking for _fseeki64 - found
  -- Looking for fstat64
  -- Looking for fstat64 - not found
  -- Looking for _fstati64
  -- Looking for _fstati64 - found
  -- Looking for fileno
  -- Looking for fileno - found
  -- Looking for _fileno
  -- Looking for _fileno - found
  -- Looking for stdint.h
  -- Looking for stdint.h - found
  -- Looking for stddef.h
  -- Looking for stddef.h - found
  -- Check size of long
  -- Check size of long - done
  -- Check size of long long
  -- Check size of long long - done
  -- Check size of off_t
  -- Check size of off_t - done
  -- Check size of off64_t
  -- Check size of off64_t - failed
  -- Check size of size_t
  -- Check size of size_t - done
  -- Check size of unsigned int
  -- Check size of unsigned int - done
  -- Check size of unsigned long
  -- Check size of unsigned long - done
  -- Check size of unsigned short
  -- Check size of unsigned short - done
  -- Compiling to Win32 - printf "%zu" size_t formatting support disabled
  -- System is little-endian.
  -- PROJECT_NAME  = librsync
  -- BUILD_HOSTNAME = DESKTOP-P725H9N
  -- CMAKE_SYSTEM = Windows-10.0.17763
  -- Could NOT find POPT (missing: POPT_LIBRARIES POPT_INCLUDE_DIRS)
  -- Could NOT find BZip2 (missing: BZIP2_LIBRARIES) (found version "1.0.6")
  -- Found ZLIB: C:/ProgramData/Miniconda3/Library/lib/z.lib (found version "1.2.11")
  -- Found components for ZLIB
  -- ZLIB_INCLUDE_DIRS  = C:/ProgramData/Miniconda3/Library/include
  -- ZLIB_LIBRARIES = C:/ProgramData/Miniconda3/Library/lib/z.lib
  -- Could NOT find LIBB2 (missing: LIBB2_LIBRARIES LIBB2_INCLUDE_DIRS)
  -- Using included blake2 implementation.
  -- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE)
  -- CMAKE_C_FLAGS  =  -nologo -MD -Brepro -Wno-format
  -- Configuring done
  -- Generating done
  -- Build files have been written to: C:/Users/admin/Documents/GitHub/librsync/gen/target/debug/build/librsync-gen-02a144ff907295ec/out/build
  running: "cmake" "--build" "." "--target" "install" "--config" "Debug" "--"
  Microsoft (R) Build Engine version 16.10.2+857e5a733 for .NET Framework
  Copyright (C) Microsoft Corporation. All rights reserved.

    Checking Build System
    Building Custom Rule C:/Users/admin/Documents/GitHub/librsync/CMakeLists.txt
  cl : command line error D8021: invalid numeric argument '/Wno-format' [C:\Users\admin\Documents\GitHub\librsync\gen\target\debug\build\librsync-gen-02a144ff907295ec\out\build\checksum_test.vcxproj]
    Building Custom Rule C:/Users/admin/Documents/GitHub/librsync/CMakeLists.txt
  cl : command line error D8021: invalid numeric argument '/Wno-format' [C:\Users\admin\Documents\GitHub\librsync\gen\target\debug\build\librsync-gen-02a144ff907295ec\out\build\hashtable_test.vcxproj]
    Building Custom Rule C:/Users/admin/Documents/GitHub/librsync/CMakeLists.txt
  cl : command line error D8021: invalid numeric argument '/Wno-format' [C:\Users\admin\Documents\GitHub\librsync\gen\target\debug\build\librsync-gen-02a144ff907295ec\out\build\isprefix_test.vcxproj]
    Building Custom Rule C:/Users/admin/Documents/GitHub/librsync/CMakeLists.txt
  cl : command line error D8021: invalid numeric argument '/Wno-format' [C:\Users\admin\Documents\GitHub\librsync\gen\target\debug\build\librsync-gen-02a144ff907295ec\out\build\netint_test.vcxproj]
    Building Custom Rule C:/Users/admin/Documents/GitHub/librsync/CMakeLists.txt
  cl : command line error D8021: invalid numeric argument '/Wno-format' [C:\Users\admin\Documents\GitHub\librsync\gen\target\debug\build\librsync-gen-02a144ff907295ec\out\build\rabinkarp_test.vcxproj]
    Building Custom Rule C:/Users/admin/Documents/GitHub/librsync/CMakeLists.txt
  cl : command line error D8021: invalid numeric argument '/Wno-format' [C:\Users\admin\Documents\GitHub\librsync\gen\target\debug\build\librsync-gen-02a144ff907295ec\out\build\rollsum_test.vcxproj]
    Building Custom Rule C:/Users/admin/Documents/GitHub/librsync/CMakeLists.txt
  cl : command line error D8021: invalid numeric argument '/Wno-format' [C:\Users\admin\Documents\GitHub\librsync\gen\target\debug\build\librsync-gen-02a144ff907295ec\out\build\rsync.vcxproj]
    Building Custom Rule C:/Users/admin/Documents/GitHub/librsync/CMakeLists.txt
  cl : command line error D8021: invalid numeric argument '/Wno-format' [C:\Users\admin\Documents\GitHub\librsync\gen\target\debug\build\librsync-gen-02a144ff907295ec\out\build\sumset_test.vcxproj]

  --- stderr
  CMake Deprecation Warning at CMakeLists.txt:20 (cmake_minimum_required):
    Compatibility with CMake < 2.8.12 will be removed from a future version of
    CMake.

    Update the VERSION argument <min> value or use a ...<max> suffix to tell
    CMake that the project does not need compatibility with older versions.


  CMake Warning (dev) at C:/Program Files/CMake/share/cmake-3.21/Modules/FindPackageHandleStandardArgs.cmake:438 (message):
    The package name passed to `find_package_handle_standard_args` (LIBB2) does
    not match the name of the calling package (libb2).  This can lead to
    problems in calling code that expects `find_package` result variables
    (e.g., `_FOUND`) to follow a certain pattern.
  Call Stack (most recent call first):
    cmake/Findlibb2.cmake:12 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
    CMakeLists.txt:177 (find_package)
  This warning is for project developers.  Use -Wno-dev to suppress it.

  CMake Warning:
    Manually-specified variables were not used by the project:

      CMAKE_ASM_FLAGS
      CMAKE_ASM_FLAGS_DEBUG
      CMAKE_CXX_FLAGS
      CMAKE_CXX_FLAGS_DEBUG


  thread 'main' panicked at '
  command did not execute successfully, got: exit code: 1

  build script failed, must exit now', C:\Users\admin\.cargo\registry\src\github.com-1ecc6299db9ec823\cmake-0.1.45\src\lib.rs:894:5
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by changhe3** [comment]
If I delete the `-Wno-format` in CmakeLists.txt, I get this:
```
PS C:\Users\admin\Documents\GitHub\librsync\gen> cargo run
    Updating crates.io index
   Compiling cc v1.0.69
   Compiling cmake v0.1.45
   Compiling librsync-gen v0.1.0 (C:\Users\admin\Documents\GitHub\librsync\gen)
error: failed to run custom build command for `librsync-gen v0.1.0 (C:\Users\admin\Documents\GitHub\librsync\gen)`

Caused by:
  process didn't exit successfully: `C:\Users\admin\Documents\GitHub\librsync\gen\target\debug\build\librsync-gen-876be86ab9e95625\build-script-build` (exit code: 101)
  --- stdout
  running: "cmake" "C:\\Users\\admin\\Documents\\GitHub\\librsync\\gen\\.." "-G" "Visual Studio 16 2019" "-Thost=x64" "-Ax64" "-DBUILD_RDIFF=OFF" "-DENABLE_COMPRESSION=OFF" "-DCMAKE_INSTALL_PREFIX=C:\\Users\\admin\\Documents\\GitHub\\librsync\\gen\\target\\debug\\build\\librsync-gen-02a144ff907295ec\\out" "-DCMAKE_C_FLAGS= -nologo -MD -Brepro" "-DCMAKE_C_FLAGS_DEBUG= -nologo -MD -Brepro" "-DCMAKE_CXX_FLAGS= -nologo -MD -Brepro" "-DCMAKE_CXX_FLAGS_DEBUG= -nologo -MD -Brepro" "-DCMAKE_ASM_FLAGS= -nologo -MD -Brepro" "-DCMAKE_ASM_FLAGS_DEBUG= -nologo -MD -Brepro" "-DCMAKE_BUILD_TYPE=Debug"
  -- Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.17763.
  -- The C compiler identification is MSVC 19.29.30040.0
  -- Detecting C compiler ABI info
  -- Detecting C compiler ABI info - done
  -- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/BuildTools/VC/Tools/MSVC/14.29.30037/bin/Hostx64/x64/cl.exe - skipped
  -- Detecting C compile features
  -- Detecting C compile features - done
  -- DO_RS_TRACE=0
  -- Looking for include file sys/file.h
  -- Looking for include file sys/file.h - not found
  -- Looking for include file sys/stat.h
  -- Looking for include file sys/stat.h - found
  -- Looking for include file sys/types.h
  -- Looking for include file sys/types.h - found
  -- Looking for include file unistd.h
  -- Looking for include file unistd.h - not found
  -- Looking for include file fcntl.h
  -- Looking for include file fcntl.h - found
  -- Looking for include file mcheck.h
  -- Looking for include file mcheck.h - not found
  -- Looking for include file zlib.h
  -- Looking for include file zlib.h - not found
  -- Looking for include file bzlib.h
  -- Looking for include file bzlib.h - not found
  -- Looking for __func__
  -- Looking for __func__ - found
  -- Looking for __FUNCTION__
  -- Looking for __FUNCTION__ - found
  -- Looking for fseeko
  -- Looking for fseeko - not found
  -- Looking for fseeko64
  -- Looking for fseeko64 - not found
  -- Looking for _fseeki64
  -- Looking for _fseeki64 - found
  -- Looking for fstat64
  -- Looking for fstat64 - not found
  -- Looking for _fstati64
  -- Looking for _fstati64 - found
  -- Looking for fileno
  -- Looking for fileno - found
  -- Looking for _fileno
  -- Looking for _fileno - found
  -- Looking for stdint.h
  -- Looking for stdint.h - found
  -- Looking for stddef.h
  -- Looking for stddef.h - found
  -- Check size of long
  -- Check size of long - done
  -- Check size of long long
  -- Check size of long long - done
  -- Check size of off_t
  -- Check size of off_t - done
  -- Check size of off64_t
  -- Check size of off64_t - failed
  -- Check size of size_t
  -- Check size of size_t - done
  -- Check size of unsigned int
  -- Check size of unsigned int - done
  -- Check size of unsigned long
  -- Check size of unsigned long - done
  -- Check size of unsigned short
  -- Check size of unsigned short - done
  -- Compiling to Win32 - printf "%zu" size_t formatting support disabled
  -- System is little-endian.
  -- PROJECT_NAME  = librsync
  -- BUILD_HOSTNAME = DESKTOP-P725H9N
  -- CMAKE_SYSTEM = Windows-10.0.17763
  -- Could NOT find POPT (missing: POPT_LIBRARIES POPT_INCLUDE_DIRS)
  -- Could NOT find BZip2 (missing: BZIP2_LIBRARIES) (found version "1.0.6")
  -- Found ZLIB: C:/ProgramData/Miniconda3/Library/lib/z.lib (found version "1.2.11")
  -- Found components for ZLIB
  -- ZLIB_INCLUDE_DIRS  = C:/ProgramData/Miniconda3/Library/include
  -- ZLIB_LIBRARIES = C:/ProgramData/Miniconda3/Library/lib/z.lib
  -- Could NOT find LIBB2 (missing: LIBB2_LIBRARIES LIBB2_INCLUDE_DIRS)
  -- Using included blake2 implementation.
  -- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE)
  -- CMAKE_C_FLAGS  =  -nologo -MD -Brepro -Wno-format
  -- Configuring done
  -- Generating done
  -- Build files have been written to: C:/Users/admin/Documents/GitHub/librsync/gen/target/debug/build/librsync-gen-02a144ff907295ec/out/build
  running: "cmake" "--build" "." "--target" "install" "--config" "Debug" "--"
  Microsoft (R) Build Engine version 16.10.2+857e5a733 for .NET Framework
  Copyright (C) Microsoft Corporation. All rights reserved.

    Checking Build System
    Building Custom Rule C:/Users/admin/Documents/GitHub/librsync/CMakeLists.txt
  cl : command line error D8021: invalid numeric argument '/Wno-format' [C:\Users\admin\Documents\GitHub\librsync\gen\target\debug\build\librsync-gen-02a144ff907295ec\out\build\checksum_test.vcxproj]
    Building Custom Rule C:/Users/admin/Documents/GitHub/librsync/CMakeLists.txt
  cl : command line error D8021: invalid numeric argument '/Wno-format' [C:\Users\admin\Documents\GitHub\librsync\gen\target\debug\build\librsync-gen-02a144ff907295ec\out\build\hashtable_test.vcxproj]
    Building Custom Rule C:/Users/admin/Documents/GitHub/librsync/CMakeLists.txt
  cl : command line error D8021: invalid numeric argument '/Wno-format' [C:\Users\admin\Documents\GitHub\librsync\gen\target\debug\build\librsync-gen-02a144ff907295ec\out\build\isprefix_test.vcxproj]
    Building Custom Rule C:/Users/admin/Documents/GitHub/librsync/CMakeLists.txt
  cl : command line error D8021: invalid numeric argument '/Wno-format' [C:\Users\admin\Documents\GitHub\librsync\gen\target\debug\build\librsync-gen-02a144ff907295ec\out\build\netint_test.vcxproj]
    Building Custom Rule C:/Users/admin/Documents/GitHub/librsync/CMakeLists.txt
  cl : command line error D8021: invalid numeric argument '/Wno-format' [C:\Users\admin\Documents\GitHub\librsync\gen\target\debug\build\librsync-gen-02a144ff907295ec\out\build\rabinkarp_test.vcxproj]
    Building Custom Rule C:/Users/admin/Documents/GitHub/librsync/CMakeLists.txt
  cl : command line error D8021: invalid numeric argument '/Wno-format' [C:\Users\admin\Documents\GitHub\librsync\gen\target\debug\build\librsync-gen-02a144ff907295ec\out\build\rollsum_test.vcxproj]
    Building Custom Rule C:/Users/admin/Documents/GitHub/librsync/CMakeLists.txt
  cl : command line error D8021: invalid numeric argument '/Wno-format' [C:\Users\admin\Documents\GitHub\librsync\gen\target\debug\build\librsync-gen-02a144ff907295ec\out\build\rsync.vcxproj]
    Building Custom Rule C:/Users/admin/Documents/GitHub/librsync/CMakeLists.txt
  cl : command line error D8021: invalid numeric argument '/Wno-format' [C:\Users\admin\Documents\GitHub\librsync\gen\target\debug\build\librsync-gen-02a144ff907295ec\out\build\sumset_test.vcxproj]

  --- stderr
  CMake Deprecation Warning at CMakeLists.txt:20 (cmake_minimum_required):
    Compatibility with CMake < 2.8.12 will be removed from a future version of
    CMake.

    Update the VERSION argument <min> value or use a ...<max> suffix to tell
    CMake that the project does not need compatibility with older versions.


  CMake Warning (dev) at C:/Program Files/CMake/share/cmake-3.21/Modules/FindPackageHandleStandardArgs.cmake:438 (message):
    The package name passed to `find_package_handle_standard_args` (LIBB2) does
    not match the name of the calling package (libb2).  This can lead to
    problems in calling code that expects `find_package` result variables
    (e.g., `_FOUND`) to follow a certain pattern.
  Call Stack (most recent call first):
    cmake/Findlibb2.cmake:12 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
    CMakeLists.txt:177 (find_package)
  This warning is for project developers.  Use -Wno-dev to suppress it.

  CMake Warning:
    Manually-specified variables were not used by the project:

      CMAKE_ASM_FLAGS
      CMAKE_ASM_FLAGS_DEBUG
      CMAKE_CXX_FLAGS
      CMAKE_CXX_FLAGS_DEBUG


  thread 'main' panicked at '
  command did not execute successfully, got: exit code: 1

  build script failed, must exit now', C:\Users\admin\.cargo\registry\src\github.com-1ecc6299db9ec823\cmake-0.1.45\src\lib.rs:894:5
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
PS C:\Users\admin\Documents\GitHub\librsync\gen> cargo run
   Compiling librsync-gen v0.1.0 (C:\Users\admin\Documents\GitHub\librsync\gen)
error: failed to run custom build command for `librsync-gen v0.1.0 (C:\Users\admin\Documents\GitHub\librsync\gen)`

Caused by:
  process didn't exit successfully: `C:\Users\admin\Documents\GitHub\librsync\gen\target\debug\build\librsync-gen-876be86ab9e95625\build-script-build` (exit code: 101)
  --- stdout
  running: "cmake" "C:\\Users\\admin\\Documents\\GitHub\\librsync\\gen\\.." "-G" "Visual Studio 16 2019" "-Thost=x64" "-Ax64" "-DBUILD_RDIFF=OFF" "-DENABLE_COMPRESSION=OFF" "-DCMAKE_INSTALL_PREFIX=C:\\Users\\admin\\Documents\\GitHub\\librsync\\gen\\target\\debug\\build\\librsync-gen-02a144ff907295ec\\out" "-DCMAKE_C_FLAGS= -nologo -MD -Brepro" "-DCMAKE_C_FLAGS_DEBUG= -nologo -MD -Brepro" "-DCMAKE_CXX_FLAGS= -nologo -MD -Brepro" "-DCMAKE_CXX_FLAGS_DEBUG= -nologo -MD -Brepro" "-DCMAKE_ASM_FLAGS= -nologo -MD -Brepro" "-DCMAKE_ASM_FLAGS_DEBUG= -nologo -MD -Brepro" "-DCMAKE_BUILD_TYPE=Debug"
  -- Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.17763.
  -- DO_RS_TRACE=0
  -- Compiling to Win32 - printf "%zu" size_t formatting support disabled
  -- System is little-endian.
  -- PROJECT_NAME  = librsync
  -- BUILD_HOSTNAME = DESKTOP-P725H9N
  -- CMAKE_SYSTEM = Windows-10.0.17763
  -- Could NOT find POPT (missing: POPT_LIBRARIES POPT_INCLUDE_DIRS)
  -- Could NOT find BZip2 (missing: BZIP2_LIBRARIES) (found version "1.0.6")
  -- Found components for ZLIB
  -- ZLIB_INCLUDE_DIRS  = C:/ProgramData/Miniconda3/Library/include
  -- ZLIB_LIBRARIES = C:/ProgramData/Miniconda3/Library/lib/z.lib
  -- Could NOT find LIBB2 (missing: LIBB2_LIBRARIES LIBB2_INCLUDE_DIRS)
  -- Using included blake2 implementation.
  -- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE)
  -- CMAKE_C_FLAGS  =  -nologo -MD -Brepro
  -- Configuring done
  -- Generating done
  -- Build files have been written to: C:/Users/admin/Documents/GitHub/librsync/gen/target/debug/build/librsync-gen-02a144ff907295ec/out/build
  running: "cmake" "--build" "." "--target" "install" "--config" "Debug" "--"
  Microsoft (R) Build Engine version 16.10.2+857e5a733 for .NET Framework
  Copyright (C) Microsoft Corporation. All rights reserved.

    Checking Build System
    Building Custom Rule C:/Users/admin/Documents/GitHub/librsync/CMakeLists.txt
    checksum_test.c
    checksum.c
    rollsum.c
    mdfour.c
    blake2b-ref.c
    Generating Code...
    checksum_test.vcxproj -> C:\Users\admin\Documents\GitHub\librsync\gen\target\debug\build\librsync-gen-02a144ff907295ec\out\build\Debug\checksum_test.exe
    Building Custom Rule C:/Users/admin/Documents/GitHub/librsync/CMakeLists.txt
    hashtable_test.c
    hashtable.c
    Generating Code...
    hashtable_test.vcxproj -> C:\Users\admin\Documents\GitHub\librsync\gen\target\debug\build\librsync-gen-02a144ff907295ec\out\build\Debug\hashtable_test.exe
    Building Custom Rule C:/Users/admin/Documents/GitHub/librsync/CMakeLists.txt
    isprefix_test.c
    isprefix.c
    Generating Code...
    isprefix_test.vcxproj -> C:\Users\admin\Documents\GitHub\librsync\gen\target\debug\build\librsync-gen-02a144ff907295ec\out\build\Debug\isprefix_test.exe
    Building Custom Rule C:/Users/admin/Documents/GitHub/librsync/CMakeLists.txt
    netint_test.c
    netint.c
    util.c
    trace.c
  C:\Users\admin\Documents\GitHub\librsync\src\trace.c(66,1): error C2491: 'rs_trace_to': definition of dllimport function not allowed [C:\Users\admin\Documents\GitHub\librsync\gen\target\debug\build\librsync-gen-02a144ff907295ec\out\build\netint_test.vcxproj]
  C:\Users\admin\Documents\GitHub\librsync\src\trace.c(71,1): error C2491: 'rs_trace_set_level': definition of dllimport function not allowed [C:\Users\admin\Documents\GitHub\librsync\gen\target\debug\build\librsync-gen-02a144ff907295ec\out\build\netint_test.vcxproj]
  C:\Users\admin\Documents\GitHub\librsync\src\trace.c(106,1): error C2491: 'rs_trace_stderr': definition of dllimport function not allowed [C:\Users\admin\Documents\GitHub\librsync\gen\target\debug\build\librsync-gen-02a144ff907295ec\out\build\netint_test.vcxproj]
  C:\Users\admin\Documents\GitHub\librsync\src\trace.c(111,1): error C2491: 'rs_supports_trace': definition of dllimport function not allowed [C:\Users\admin\Documents\GitHub\librsync\gen\target\debug\build\librsync-gen-02a144ff907295ec\out\build\netint_test.vcxproj]
    tube.c
    scoop.c
    stream.c
    Generating Code...
    Building Custom Rule C:/Users/admin/Documents/GitHub/librsync/CMakeLists.txt
    rabinkarp_test.c
    rabinkarp_test.vcxproj -> C:\Users\admin\Documents\GitHub\librsync\gen\target\debug\build\librsync-gen-02a144ff907295ec\out\build\Debug\rabinkarp_test.exe
    Building Custom Rule C:/Users/admin/Documents/GitHub/librsync/CMakeLists.txt
    rollsum_test.c
    rollsum.c
    Generating Code...
    rollsum_test.vcxproj -> C:\Users\admin\Documents\GitHub\librsync\gen\target\debug\build\librsync-gen-02a144ff907295ec\out\build\Debug\rollsum_test.exe
    Building Custom Rule C:/Users/admin/Documents/GitHub/librsync/CMakeLists.txt
    prototab.c
    base64.c
    buf.c
    checksum.c
    command.c
    delta.c
    emit.c
    fileutil.c
    hashtable.c
    hex.c
    job.c
    mdfour.c
    mksum.c
    msg.c
    netint.c
    patch.c
    readsums.c
    rollsum.c
    scoop.c
    stats.c
    Generating Code...
    Compiling...
    stream.c
    sumset.c
    trace.c
    tube.c
    util.c
    version.c
    whole.c
    blake2b-ref.c
    Generating Code...
       Creating library C:/Users/admin/Documents/GitHub/librsync/gen/target/debug/build/librsync-gen-02a144ff907295ec/out/build/Debug/rsync.lib and object C:/Users/admin/Documents/GitHub/librsync/gen/target/debug/build/librsync-gen-02a144ff907295ec/out/build/Debug/rsync.exp
    rsync.vcxproj -> C:\Users\admin\Documents\GitHub\librsync\gen\target\debug\build\librsync-gen-02a144ff907295ec\out\build\Debug\rsync.dll
    Building Custom Rule C:/Users/admin/Documents/GitHub/librsync/CMakeLists.txt
    sumset_test.c
    sumset.c
    util.c
    trace.c
    hex.c
    checksum.c
    rollsum.c
    mdfour.c
    hashtable.c
    blake2b-ref.c
    Generating Code...
    sumset_test.vcxproj -> C:\Users\admin\Documents\GitHub\librsync\gen\target\debug\build\librsync-gen-02a144ff907295ec\out\build\Debug\sumset_test.exe

  --- stderr
  CMake Deprecation Warning at CMakeLists.txt:20 (cmake_minimum_required):
    Compatibility with CMake < 2.8.12 will be removed from a future version of
    CMake.

    Update the VERSION argument <min> value or use a ...<max> suffix to tell
    CMake that the project does not need compatibility with older versions.


  CMake Warning (dev) at C:/Program Files/CMake/share/cmake-3.21/Modules/FindPackageHandleStandardArgs.cmake:438 (message):
    The package name passed to `find_package_handle_standard_args` (LIBB2) does
    not match the name of the calling package (libb2).  This can lead to
    problems in calling code that expects `find_package` result variables
    (e.g., `_FOUND`) to follow a certain pattern.
  Call Stack (most recent call first):
    cmake/Findlibb2.cmake:12 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
    CMakeLists.txt:177 (find_package)
  This warning is for project developers.  Use -Wno-dev to suppress it.

  thread 'main' panicked at '
  command did not execute successfully, got: exit code: 1

  build script failed, must exit now', C:\Users\admin\.cargo\registry\src\github.com-1ecc6299db9ec823\cmake-0.1.45\src\lib.rs:894:5
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by mbrt** [comment]
Fixed by #22.
